src/librustc/ich/impls_ty.rs

```diff
@@ -385,10 +392,15 @@ impl<'a> HashStable<StableHashingContext<'a>> for mir::interpret::Allocation {
         }
         self.undef_mask.hash_stable(hcx, hasher);
         self.align.hash_stable(hcx, hasher);
-        self.mutable.hash_stable(hcx, hasher);
+        self.runtime_mutability.hash_stable(hcx, hasher);
     }
 }
 
+impl_stable_hash_for!(enum ::syntax::ast::Mutability {
```

Comment:
Isn't there a type like `mir::Mutability` or similar that implements stable hash already?

Reply:
Probably. There are three of these enums. Any reason not to merge them all into one?

Reply:
Ask @nikomatsakis.